dtfs-central-api/src/helpers/amendments/filter-and-sort-completed-effective-amendments.ts

```diff
@@ -0,0 +1,36 @@
+import { AMENDMENT_STATUS, isValidDate, TfmFacilityAmendment } from '@ukef/dtfs2-common';
+import { isBefore } from 'date-fns';
+import { orderBy } from 'lodash';
+
+/**
+ * Filters amendments to only those which are completed
+ * and effective from or before the given date.
+ *
+ * Sorts the filtered amendments by effective date descending.
+ *
+ * If no effective date is present then uses the updatedAt
+ * date in it's place as a fallback.
+ * @param tfmFacilityAmendments - amendments to filter
+ * @param latestEffectiveDate - The latest effective date to consider
+ * @returns completed amendments
+ */
+export const filterAndSortCompletedEffectiveAmendments = (tfmFacilityAmendments: TfmFacilityAmendment[], latestEffectiveDate: Date): TfmFacilityAmendment[] => {
+  const filteredAmendments = tfmFacilityAmendments.filter(({ status, effectiveDate, updatedAt, amendmentId }) => {
+    if (status !== AMENDMENT_STATUS.COMPLETED) {
+      return false;
+    }
+
+    const dateToUse = effectiveDate ? new Date(effectiveDate) : new Date(updatedAt);
+
+    if (!isValidDate(dateToUse)) {
+      const invalidField = effectiveDate ? 'effective date' : 'updated at';
```

Comment:
I'd prefer these to be real camelCase fields, what do you think?

Reply:
Agreed, I think that is better!